gravitee-apim-portal-webui-next/src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,116 @@
+import { MatIcon } from '@angular/material/icon';
+import { MatError, MatFormField, MatInput, MatLabel } from '@angular/material/input';
+import { MatOption, MatSelect } from '@angular/material/select';
+import { RouterLink } from '@angular/router';
+import { tap } from 'rxjs';
+
+import { MobileClassDirective } from '../../directives/mobile-class.directive';
+import { CustomUserFields } from '../../entities/user/custom-user-fields';
+import { UsersService } from '../../services/users.service';
+
+interface UserRegistrationFormValue {
+  firstname: string;
+  lastname: string;
+  email: string;
+  customFields?: { [key: string]: string };
+}
+
+@Component({
+  selector: 'app-registration',
+  imports: [
+    MatCard,
+    MatCardContent,
+    MatCardHeader,
+    MatCardTitle,
+    MatError,
+    MatFormField,
+    MatInput,
+    MatLabel,
+    MatButtonModule,
+    ReactiveFormsModule,
+    RouterLink,
+    MobileClassDirective,
+    MatError,
+    MatFormField,
+    MatOption,
+    MatSelect,
+    MatIcon,
+  ],
+  templateUrl: './registration.component.html',
+  styleUrl: './registration.component.scss',
+})
+export class RegistrationComponent implements OnInit {
+  submitted: boolean = false;
+  sentToEmail = '';
+
+  customUserFields: CustomUserFields[] = [];
+
+  registrationForm: FormGroup<{
+    firstname: FormControl;
+    lastname: FormControl;
+    email: FormControl;
+    customFields?: FormGroup;
+  }> = new FormGroup({
+    firstname: new FormControl('', [Validators.required]),
+    lastname: new FormControl('', [Validators.required]),
+    email: new FormControl('', [Validators.required, Validators.email]),
+  });
+
+  error = signal(200);
+
+  constructor(
+    private usersService: UsersService,
+    private readonly destroyRef: DestroyRef,
+  ) {}
+
+  ngOnInit(): void {
+    this.usersService
+      .listCustomUserFields()
+      .pipe(
+        tap(customUserFields => {
+          this.customUserFields = customUserFields;
+          if ((customUserFields || []).length > 0) {
+            const customFields = new FormGroup({});
+            for (const customUserField of customUserFields) {
+              customFields.addControl(customUserField.key!, new FormControl('', customUserField.required ? [Validators.required] : []));
+            }
+            this.registrationForm.addControl('customFields', customFields);
+          }
+        }),
+      )
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe();
+  }
+
+  register() {
+    const confirmationPageUrl = window.location.origin + window.location.pathname + '/confirm';
+    const userRegistrationFormValue = this.registrationForm.value as UserRegistrationFormValue;
+
+    this.usersService
+      .registerNewUser({ ...userRegistrationFormValue, confirmation_page_url: confirmationPageUrl })
+
+      .pipe(tap(() => (this.sentToEmail = userRegistrationFormValue.email)))
+      .pipe(tap(() => (this.submitted = true)))
```

Comment:
Related to https://github.com/gravitee-io/gravitee-api-management/pull/14707/changes#r2640097808

```suggestion
      .pipe(tap(() => {
        this.sentToEmail.set(userRegistrationFormValue.email);
        this.submitted.set(true);
      }))